Lib/test/test_context.py

```diff
@@ -360,6 +361,62 @@ def sub(num):
             tp.shutdown()
         self.assertEqual(results, list(range(10)))
 
+    @isolated_context
+    def test_context_manager_1(self):
+        cvar = contextvars.ContextVar('cvar', default='initial')
+        self.assertEqual(cvar.get(), 'initial')
+        ctx = contextvars.copy_context()
+        with ctx as tmp:
+            self.assertIs(tmp, ctx)
+            self.assertEqual(cvar.get(), 'initial')
+            cvar.set('updated')
+            self.assertEqual(cvar.get(), 'updated')
+        self.assertEqual(cvar.get(), 'initial')
+        self.assertEqual(ctx[cvar], 'updated')
+        with ctx:
+            self.assertEqual(cvar.get(), 'updated')
+
+    @threading_helper.requires_working_threading()
+    def test_context_manager_2(self):
+        ctx = contextvars.copy_context()
+        thread = threading.Thread(target=ctx.__enter__)
```

Comment:
I think it won't hurt to add a thread-safety test, something like `test_context_threads_1`. How hard would it be?

Reply:
I don't quite understand what `test_context_threads_1` is meant to test.  Maybe it's testing that each thread automatically gets its own Context instance?  If that wasn't true then the test would fail with high probability, I think.

If that test is meant to test internal state data races then I don't understand how it accomplishes that goal, as sleeping for even 1ms significantly drops the probability of losing a data race.

It would be great to write a test that guarantees that concurrent `Context.__enter__()` and/or `Context.run()` calls on the same Context object will result in exactly one successful call, and `RuntimeError` exceptions for the other calls, without any internal state corruption due to data races.  I'm not sure how to write such a test.

However, I'm not sure it's worth adding such a test in this PR:
  * The possibility of concurrent enters is not new to this PR (one can call `Context.run()` concurrently today).
  * This PR doesn't make concurrent enters more likely.
  * Users are unlikely to encounter a race because Context objects are not meant to be used across threads.  (There is no way to hand off ownership of an entered Context to another thread, and exiting a Context from a different thread will raise.)
  * I think the test would fail because `_PyContext_Enter` does not atomically check and set `ctx->ctx_entered`.  Fixing that feels out of scope for this PR.

It might be worth addressing thread safety in another PR, but as far as I know nobody has complained so far, so maybe it's better to just leave it alone.